### ISSUE TITLE
Latest Thanos Update breaks the library

### ISSUE THREAD

**Issue by jan-kantert** [puetzp/prometheus-http-query#15]
We use this library to query thanos. Recent versions added a new entry to the json response:
```json
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {},
        "value": [
          1733333548.697,
          "0"
        ]
      }
    ],
    "stats": {
      "samples": {
        "totalQueryableSamples": 63,
        "totalQueryableSamplesPerStep": null,
        "peakSamples": 18
      }
    },
    "analysis": {
      "name": "",
      "executionTime": "0s",
      "children": null
    }
  },
  "warnings": [
    "PromQL info: metric might not be a counter, name does not end in _total/_sum/_count/_bucket: \"nginx_ingress_controller_requests\""
  ]
}
```
This results in this error inside the library:
```
failed to parse JSON response from server
```

**Comment by jan-kantert** [comment]
This starts with thanos v0.37.0 which merged https://github.com/thanos-io/thanos/pull/7852. prometheus-http-query expects a timing field: `Error: missing field "timings"`.

**Comment by puetzp** [comment]
Hey there,

I would like to wait how the discussion on the linked issue plays out. Seems like the easiest fix would be to make the `timings` field in `stats` optional (which it currently is not) and default to `None` when it is not provided in the response returned by the server. 

However I am hesitant to change the API when there is still a possibility that the Thanos project might fix it on their end to ensure Prometheus API compatibility. I am unsure how likely that is given I that I am not well acquainted with the Thanos project.

Please let me know if you disagree. Otherwise I am sure to keep an eye on the issue but wait a little bit before providing a solution.

**Comment by jan-kantert** [comment]
> Hey there,
> 
> I would like to wait how the discussion on the linked issue plays out. Seems like the easiest fix would be to make the `timings` field in `stats` optional (which it currently is not) and default to `None` when it is not provided in the response returned by the server.
> 
> However I am hesitant to change the API when there is still a possibility that the Thanos project might fix it on their end to ensure Prometheus API compatibility. I am unsure how likely that is given I that I am not well acquainted with the Thanos project.
> 
> Please let me know if you disagree. Otherwise I am sure to keep an eye on the issue but wait a little bit before providing a solution.

Makes sense. I currently locally changed the struct to contain an option:
```
pub struct Stats {
    timings: Option<Timings>,
    samples: Option<Samples>,
}
```
Works fine but would be a breaking API change for everyone using `stats`.
